File: DKPlugins/DKDuktape/DKUIEvent.js
// https://developer.mozilla.org/en-US/docs/Web/API/UIEvent

var UIEvent = function UIEvent(_type, _options, _pointer) {
	//console.log("UIEvent("+_type+","+_options+","+_pointer+")")
	
	/*
	this._type = _type;
	this._options = _options;
	this._pointer = _pointer;
	*/
	
    //// Properties ////
    Object.defineProperty(this, "detail", {
        //Read only
        get: function detail() {
            return CPP_DKUIEvent_detail(_pointer);
        }
    })
    Object.defineProperty(this, "sourceCapabilities", {
        //Read only
        get: function sourceCapabilities () {
            return CPP_DKUIEvent_sourceCapabilities(_pointer);
        }
    })
	Object.defineProperty(this, "view", {
        //Read only
        get: function view() {
            return CPP_DKUIEvent_view(_pointer);
        }
    })
	Object.defineProperty(this, "which", {
        //Read only
        get: function which() {
            return CPP_DKUIEvent_which(_pointer);
        }
    })

    //// Methods ////
    UIEvent.prototype.initUIEvent = function initUIEvent() {
        //Deprecated
        CPP_DKUIEvent_initUIEvent(this._pointer);
    }

	//// toString ////
	if(this.toString() === "[object Object]"){
		this.toString = function(){
			return "[object UIEvent]"
		}
	}
	
    return Event.call(this, _type, _options, _pointer);
};
UIEvent.prototype = Event.prototype;

File: DKPlugins/DKDuktape/DKUIEvent.cpp
// https://developer.mozilla.org/en-US/docs/Web/API/UIEvent

#include "DK/DKApp.h"
#include "DKDuktape/DKUIEvent.h"
#include "DKDuktape/DKEvent.h"
//#include "DKRml/DKRml.h"


bool DKUIEvent::Init(){
	DKDEBUGFUNC();

	// Properties
	//DKDuktape::AttachFunction("CPP_DKUIEvent_cancelBubble", DKUIEvent::cancelBubble);

	// Methods
	DKDuktape::AttachFunction("CPP_DKUIEvent_initUIEvent", DKUIEvent::initUIEvent);

	DKClass::DKCreate("DKDuktape/DKUIEvent.js");
	return true;
}


// Properties
/*
int DKUIEvent::cancelBubble(duk_context* ctx){
	DKDEBUGFUNC(ctx);
	DKString eventAddress = duk_require_string(ctx, 0);
	Rml::Event* event = DKEvent::addressToEvent(eventAddress);
	if (!event) {
		DKERROR("DKUIEvent::cencelBubble(): event invalid\n");
		duk_push_boolean(ctx, false);
		return true;
	}
	bool flag = duk_require_boolean(ctx, 1);
	if(flag){
		event->StopPropagation();
		return true;
	}
	return true;
}
*/

// Methods
int DKUIEvent::initUIEvent(duk_context* ctx){
	DKDEBUGFUNC(ctx);
	/*
	DKString eventAddress = duk_require_string(ctx, 0);
	Rml::Event* event = DKRml::addressToEvent(eventAddress);
	if (!event) {
		DKERROR("DKUIEvent::initEvent(): event invalid\n");
		duk_push_boolean(ctx, false);
		return true;
	}

	//TODO
	*/
	return false;
}